Build failure: rubyfmt

### Steps To Reproduce

Happens on all platforms.

Steps to reproduce the behavior:
1. build rubyfmt

### Build log

```
...
  linking ruby
  make[2]: Leaving directory '/private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout'
  make[1]: Leaving directory '/private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout'
  make[1]: Entering directory '/private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout'
  *** Following extensions are not compiled:
  fiddle:
        Could not be configured. It will not be installed.
        /private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout/ext/fiddle/extconf.rb:73: missing libffi. Please install libffi or use --with-libffi-source-dir w>
        Check ext/fiddle/mkmf.log for more details.
  openssl:
        Could not be configured. It will not be installed.
        /private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout/ext/openssl/extconf.rb:101: OpenSSL library could not be found. You might want to use --with-open>
        Check ext/openssl/mkmf.log for more details.
  *** Fix the problems, then remove these directories and try again if you want.
  make[1]: Leaving directory '/private/tmp/nix-build-rubyfmt-0.10.0.drv-0/source/librubyfmt/ruby_checkout'

  --- stderr
  + cp enc/jis/props.h.blt enc/jis/props.h
  parse.c:9130:9: warning: variable 'yynerrs' set but not used [-Wunused-but-set-variable]
      int yynerrs = 0;
          ^
  1 warning generated.
  bigdecimal.c:7161:18: warning: variable 'prec' set but not used [-Wunused-but-set-variable]
      SIGNED_VALUE prec;
                   ^
  ripper.c:10310:9: warning: variable 'yynerrs' set but not used [-Wunused-but-set-variable]
      int yynerrs = 0;
          ^
  1 warning generated.
  1 warning generated.
  thread 'main' panicked at librubyfmt/build.rs:90:5:
  could not extract arch from rbconfig.rb
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

cc @bobvanderlinden 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.31, NixOS, 24.11 (Vicuña), 24.11.20240612.57d6973`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/dydg48djlykksz8cxq0xjplyxpa9pvf4-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/rubyfmt/0003-ignore-warnings.patch
+++ b/pkgs/development/tools/rubyfmt/0003-ignore-warnings.patch
@@ -1,3 +1,15 @@
+diff --git i/librubyfmt/build.rs w/librubyfmt/build.rs
+index 296b749..941a4ca 100644
+--- i/librubyfmt/build.rs
++++ w/librubyfmt/build.rs
+@@ -166,6 +166,7 @@ fn run_configure(ruby_checkout_path: &Path) -> Output {
+         command
+             .arg("--target=aarch64-unknown-linux-gnu")
+             .arg("--host=x86_64")
++            .arg("--disable-werror")
+             .env("CC", "aarch64-linux-gnu-gcc")
+             .env("AR", "aarch64-linux-gnu-ar")
+             .env("RANLIB", "aarch64-linux-gnu-ranlib");
 diff --git i/librubyfmt/src/lib.rs w/librubyfmt/src/lib.rs
 index 9b94b5f..b78e99f 100644
 --- i/librubyfmt/src/lib.rs
@@ -6,3 +18,4 @@ index 9b94b5f..b78e99f 100644
 -#![deny(warnings, missing_copy_implementations)]
  #![allow(clippy::upper_case_acronyms, clippy::enum_variant_names)]
  
+ use serde::de::value;

--- a/pkgs/development/tools/rubyfmt/0003-ignore-warnings.patch
+++ b/pkgs/development/tools/rubyfmt/0003-ignore-warnings.patch
@@ -1,0 +1,8 @@
+diff --git i/librubyfmt/src/lib.rs w/librubyfmt/src/lib.rs
+index 9b94b5f..b78e99f 100644
+--- i/librubyfmt/src/lib.rs
++++ w/librubyfmt/src/lib.rs
+@@ -1,4 +1,3 @@
+-#![deny(warnings, missing_copy_implementations)]
+ #![allow(clippy::upper_case_acronyms, clippy::enum_variant_names)]
+ 

--- a/pkgs/development/tools/rubyfmt/default.nix
+++ b/pkgs/development/tools/rubyfmt/default.nix
@@ -61,6 +61,8 @@ rustPlatform.buildRustPackage rec {
   cargoPatches = [
     # Avoid checking whether ruby gitsubmodule is up-to-date.
     ./0002-remove-dependency-on-git.patch
+    # Avoid failing on unused variable warnings.
+    ./0003-ignore-warnings.patch
   ];
 
   cargoHash = "sha256-QZ26GmsKyENkzdCGg2peie/aJhEt7KQAF/lwsibonDk=";
@@ -76,12 +78,7 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/fables-tales/rubyfmt";
     license = licenses.mit;
     maintainers = with maintainers; [ bobvanderlinden ];
-    # https://github.com/NixOS/nixpkgs/issues/320722
-    broken = true
-      # = note: Undefined symbols for architecture x86_64:
-      #       "_utimensat", referenced from:
-      #           _utime_internal in librubyfmt-3c969812b3b27083.rlib(file.o)
-      || stdenv.isDarwin && stdenv.isx86_64;
+    broken = stdenv.isDarwin && stdenv.isx86_64;
     mainProgram = "rubyfmt";
   };
 }

